Updated dependencies

fredrikbaberg opened the pull request:
On a clean ROS workspace (Ubuntu 14.04, ROS Indigo) am_driver_legacy does not compile due to not finding am_driver. Thus I added it as a build dependency to the package.xml file.

I've also:
* Added dependencies so that it should be enough to use `rosdep install --from-path src --ignore-src` to install dependencies.
* Removed am_driver_legacy attempting to generate messages (no messages are defined, it seems to use am_driver messages).
* Added code to ensure am_driver_safe generates service messages before trying to compile the node, which otherwise could cause catkin_make (or catkin build) to fail when it did not find the messages first.

I've also modified dependencies on am_description, am_driver and am_driver_legacy

fredrikbaberg committed:
Added am_driver as dependency on am_driver_legacy.

fredrikbaberg committed:
More missing dependencies, should be enough to use 'rosdep install' for missing dependencies.

fredrikbaberg committed:
Removed message generation from am_driver_legacy, made sure am_driver_safe generates srv first to avoid compilation error.

StefanGrufman reviewed (commented):
Hi there Fredrik!
Sorry for not looking into this pull request earlier. Sorry about that. 
I was reviewing the changes and saw you added "hector_gazebo_plugins" and "pyhton-pygame" to the am_driver package. I am thinking that this would break our idea that the "am_driver" should be possible to run on a target node without gazebo etc. Would this addition require those in order to compile/run? If so, we need to find another way (place?) of adding them.

fredrikbaberg commented:
I'll have a new look at what is needed. I just did a clean install of Ubuntu 16.04, so hopefully I can test what happens with a minimal install.

fredrikbaberg committed:
Merge branch 'master'

Conflicts:
	am_driver_safe/CMakeLists.txt

fredrikbaberg committed:
Updated dependencies, removal and documentation

Removed and updated some of the dependencies, and added documentation why some of the lines were added.

fredrikbaberg committed:
Changed urdf in launchfile

URDF model is missing from repo, so changed to another one.

fredrikbaberg commented:
I've made some changes, including making sure the pull request is up to date compared to master.

The issue of am_driver depending on hector_gazebo_plugins and pygames was because the launchfile in am_driver depended on am_description, which depended on hector_gazebo_plugins (used in the URDF for gazebo). I've removed hector_gazebo_plugins now, but pygame is still there (required by nanoKontrol2.py).

I'm not sure how you want to handle dependencies in launchfiles, as it does not break the code from compiling but does prevent the launchfile from running properly, so I put comments where dependencies related to that are added.
